json.go

```diff
@@ -0,0 +1,174 @@
+type JSONAttachment struct {
+	Url      string        `json:"url,omitempty"`
+	MIMEType string        `json:"mime_type,omitempty"`
+	Title    string        `json:"title,omitempty"`
+	Size     int32         `json:"size,omitempty"`
+	Duration time.Duration `json:"duration_in_seconds,omitempty"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+// The Duration field is marshaled in seconds, all other fields are marshaled
+// based upon the definitions in struct tags.
+func (a *JSONAttachment) MarshalJSON() ([]byte, error) {
+	type EmbeddedJSONAttachment JSONAttachment
+	return json.Marshal(&struct {
+		Duration float64 `json:"duration_in_seconds,omitempty"`
+		*EmbeddedJSONAttachment
+	}{
+		EmbeddedJSONAttachment: (*EmbeddedJSONAttachment)(a),
+		Duration:               a.Duration.Seconds(),
+	})
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+// The Duration field is expected to be in seconds, all other field types
+// match the struct definition.
+func (a *JSONAttachment) UnmarshalJSON(data []byte) error {
+	type EmbeddedJSONAttachment JSONAttachment
+	var raw struct {
+		Duration float64 `json:"duration_in_seconds,omitempty"`
+		*EmbeddedJSONAttachment
+	}
+	raw.EmbeddedJSONAttachment = (*EmbeddedJSONAttachment)(a)
+
+	err := json.Unmarshal(data, &raw)
+	if err != nil {
+		return err
+	}
+
+	if raw.Duration > 0 {
+		nsec := int64(raw.Duration * float64(time.Second))
+		raw.EmbeddedJSONAttachment.Duration = time.Duration(nsec)
+	}
+
+	return nil
+}
+
+// JSONItem represents a single entry/post for the feed.
+type JSONItem struct {
+	Id            string           `json:"id"`
+	Url           string           `json:"url,omitempty"`
+	ExternalUrl   string           `json:"external_url,omitempty"`
+	Title         string           `json:"title,omitempty"`
+	ContentHTML   string           `json:"content_html,omitempty"`
+	ContentText   string           `json:"content_text,omitempty"`
+	Summary       string           `json:"summary,omitempty"`
+	Image         string           `json:"image,omitempty"`
+	BannerImage   string           `json:"banner_,omitempty"`
+	PublishedDate *time.Time       `json:"date_published,omitempty"`
+	ModifiedDate  *time.Time       `json:"date_modified,omitempty"`
+	Author        *JSONAuthor      `json:"author,omitempty"`
+	Tags          []string         `json:"tags,omitempty"`
+	Attachments   []JSONAttachment `json:"attachments,omitempty"`
+}
+
+// JSONHub describes an endpoint that can be used to subscribe to real-time
+// notifications from the publisher of this feed.
+type JSONHub struct {
+	Type string `json:"type"`
+	Url  string `json:"url"`
+}
+
+// JSONFeed represents a syndication feed in the JSON Feed Version 1 format.
+// Matching the specification found here: https://jsonfeed.org/version/1.
+type JSONFeed struct {
+	Version     string      `json:"version"`
+	Title       string      `json:"title"`
+	HomePageUrl string      `json:"home_page_url,omitempty"`
```

Comment:
Not changed based on comment @kisielk about remaining internally consistent with other names in this package.